docs/SETUP.md

```diff
@@ -0,0 +1,142 @@
+### Installation for developers
+
+The section below is intended for developers.
+
+<summary><details>
+
+#### Version manager
+
+We recommend to use a version manager like [nvm](https://github.com/nvm-sh/nvm) or [nodenv](https://github.com/nodenv/nodenv) to automatically configure the version of Node required by this software package. The [.nvmrc](https://github.com/Automattic/vip-cli/blob/trunk/.nvmrc) file gives these tools the necessary hints for what version to use.
+
+#### Install and setup of API
+
+This software relies on https://api.wpvip.com. You may need to have a local instance of that set up for development. Follow internal instructions in that case.
+
+### Fetching & installing
+
+This will fetch the package and install all dependencies:
+
+```bash
+git clone git@github.com:Automattic/vip-cli.git && \
+cd vip-cli && npm install
+```
+
+### Building
+
+This will build all TypeScript files so they can be executed:
+
+```bash
+cd vip-cli && \
+npm run build
+```
+
+</details></summary>
+
+## Usage
+
+The software runs as standalone CLI and relies on environmental variables for configuration and a few configuration files.
+
+### Starting up locally
+
+To start the software locally, run:
+
+```bash
+vip
+```
+
+If you need more information, check out our [VIP-CLI documentation](https://docs.wpvip.com/technical-references/vip-cli/).
+
+### Analytics
+
+By default, we record information about the usage of this tool using an in-house analytics sytem. If you would prefer to opt-out of this data collection, you can do so via the `DO_NOT_TRACK` environment variable. You may either export it in your shell configuration or specify it on the command line (e.g. `DO_NOT_TRACK=1 vip app list`).
+
+### Local application
+
+#### Configuring access token
+
+Install the software locally, run and follow the instructions to configure the access token.
+
+### Environmental variables
+
+#### Configuring environmental variables
+
+Environmental variables are configured in the shell. Use normal shell commands to set them.
+
+#### List of environmental variables
+
+This application uses environmental variables for vital configuration information. Find a partial list below.
+
+TODO:
+
+- `A`: Used to ...
+- `B`: Used to ...
+
+## Updating dependencies
+
+### NPM packages
+
+Dependencies generally should be updated by merging [pull requests from dependabot](https://github.com/Automattic/vip-cli/pulls/app%2Fdependabot). However, great care should be taken before merging as updating dependencies can cause fatal errors with the `vip-cli` application, or make it unstable or misbehave. There can also be effects to the deployment process. This can happen silently.
+
+### Upgrading Node Version
+
+To upgrade Node, follow these steps:
+
+1. Update [.nvmrc](https://github.com/Automattic/vip-cli/blob/trunk/.nvmrc) to the new version and open a pull request.
+1. Address any test failures.
+1. Update the Node.js version locally and test before merging into the repository.
```

Comment:
1. Update `engines.node` in `package.json`.

Reply:
Resolved in 43a15e3.